rules/rbe_repo/checked_in.bzl

```diff
@@ -38,70 +27,265 @@ def validateUseOfCheckedInConfigs(
+              is set. Set to point to def container_to_config_spec_names()
+              defined in the versions.bzl file generated in the output_base defined
+              in the toolchain_config_suite_spec.
+          toolchain_config_specs: Must point to a list containing structs,
+              each struct represents a repo config with 'name' (str),
+             'java_home'(str), 'create_java_configs' (bool), 'create_cc_configs' (bool),
+             'config_repos' (string list) and 'env' (dict).
+          bazel_to_config_spec_names_map: Set to point to def bazel_to_config_versions()
+              defined in the versions.bzl file generated in the output_base defined
+              in the toolchain_config_suite_spec.
+      repository: The path to the toolchain container on the registry.
+      requested_toolchain_config_spec_name: the toolchain_config_spec_name of the config requested by the user.
+      tag: The tag on the toolchain container.
+      use_checked_in_confs: Whether to use checked in configs.
 
     Returns:
-        None
+      None
     """
     if use_checked_in_confs == CHECKED_IN_CONFS_FALSE:
-        return None
-    if not base_container_digest and registry and registry != RBE_UBUNTU_REGISTRY:
-        return None
-    if not base_container_digest and repository and repository != RBE_UBUNTU_REPO:
-        return None
-    if env and env != clang_env():
-        return None
-    if create_java_configs and java_home:
-        return None
+        print("%s not using checked in configs as user set attr to false " % name)
+        return None, None
     if bazel_rc_version:
-        return None
+        print("%s not using checked in configs as hazel rc version was used " % name)
+        return None, None
+
+    if not base_container_digest:
+        # If no base_container_digest was set we only use checked-in confs if
+        # the registry and repo match the toolchain_config_suite_spec
+        if registry and registry != toolchain_config_suite_spec["container_registry"]:
+            print(("%s not using checked in configs; registry was set to '%s' " +
+                   "and toolchain_config_suite_spec is configured for '%s'") %
+                  (name, registry, toolchain_config_suite_spec["container_registry"]))
+            return None, None
+        if repository and repository != toolchain_config_suite_spec["container_repo"]:
+            print(("%s not using checked in configs; repository was set to '%s' " +
+                   "and toolchain_config_suite_spec is configured for '%s'") %
+                  (name, repository, toolchain_config_suite_spec["container_repo"]))
+            return None, None
+    bazel_to_config_spec_names_map = toolchain_config_suite_spec["toolchain_config_suite_autogen_spec"].bazel_to_config_spec_names_map
+    if not bazel_to_config_spec_names_map.get(bazel_version):
+        # if Bazel version is a minor version above .0, lets try to default to the .0 config
```

Comment:
This means the toolchain configs we publish for minor releases will be completely ignored by rbe_autoconfig. Should we stop publishing configs for minor Bazel updates then?

Reply:
delayed feature to fallback to a future PR